robot/impl/local_robot_test.go

```diff
@@ -4406,12 +4409,9 @@ func TestRemovingOfflineRemotes(t *testing.T) {
 	time.Sleep(2 * time.Second)
 
 	// Ensure that the remote is not marked for removal while trying to connect to the remote
-	testutils.WaitForAssertion(t, func(tb testing.TB) {
```

Comment:
I'm slightly surprised `WaitForAssertion` didn't work, but it's more likely that I don't fully understand the guarantees of this function